chef/cookbooks/ceph/recipes/radosgw_civetweb.rb

```diff
@@ -0,0 +1,79 @@
+# This recipe doesn't do anything other than deal with SSL keys and certificates.
+# Everything else for civetweb setup is in conf.rb.  It could arguably be merged
+# into radosgw.rb; it only exists as a separate file because it used to be
+# radosgw_apache2.rb
+
+if node["ceph"]["radosgw"]["ssl"]["enabled"]
+  certfile      = node["ceph"]["radosgw"]["ssl"]["certfile"]
+  keyfile       = node["ceph"]["radosgw"]["ssl"]["keyfile"]
+  pemfile       = node["ceph"]["radosgw"]["ssl"]["pemfile"]
+  if  node["ceph"]["radosgw"]["ssl"]["generate_certs"]
+    package "openssl"
+    ruby_block "generate_certs for radosgw" do
+        block do
+          unless ::File.exists? certfile and ::File.exists? keyfile
+            require "fileutils"
+
+            Chef::Log.info("Generating SSL certificate for radosgw...")
+
+            [:certfile, :keyfile].each do |k|
+              dir = File.dirname(node[:ceph][:radosgw][:ssl][k])
+              FileUtils.mkdir_p(dir) unless File.exists?(dir)
```

Comment:
Lint/DeprecatedClassMethods: `File.exists?` is deprecated in favor of `File.exist?`.<br>Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)